### ISSUE TITLE
[Bug]:cannot run CLI or other operations behind https, since v1.12.1

### ISSUE THREAD

**Issue by kyleli666** [RedisInsight/RedisInsight#879]
**Preconditions** (Any important steps we need to know)
[RedisInsight Docker image](https://hub.docker.com/r/redislabs/redisinsight/tags) tag 1.12.1

**Steps to reproduce** (How to reproduce what you found step by step)
1. Deploy as per redisinsight document about [installation in Kubernetes](https://docs.redis.com/latest/ri/installing/install-k8s/), while the latest tag equals tag 1.12.1
2. Create an ingress of aws alb class, and config https annotations, while redisinsight is the backend
3. Browse the https url and add a redis database
4. Do add key or delete key / Turn to CLI menu to run command / change right corner settings

**Actual behavior** (A short description of what you found)
Cannot do the above Step 4 operation, request is denied with status code 403 and response "CSRF Failed: Origin checking failed ..."
Does not meet this issue with 1.11.1 or 1.12.0
![image](https://user-images.githubusercontent.com/93997542/178502842-a620bc61-5a2d-434a-b7c8-61c8f469d5dc.png)

**Expected behavior** (A short description of what you expected to find)
All operation works

**Screenshots** (Paste or drag-and-drop a screenshot or a link to a recording)

**Additional context** (Operating system, version of RedisInsight, Redis database version, Redis module version, database type, connection type, logs, or any other information)

**Comment by ewoelfel** [comment]
I also experienced this error. Reverting to 1.11.1 helped resolving the issue for now. The reason seems to be that the django framework checks the trusted origins (see https://docs.djangoproject.com/en/4.0/ref/settings/#csrf-trusted-origins)
I also tried using the new redis insight 2.0 GA which did not have the CSRF-Issue however i experienced timeouts resolving the cluster

**Comment by GnaneshKunal** [comment]
@kyleli666 

I'm looking into the issue. Can you also meanwhile check if the proxy is setup right?

Here's an nginx one: https://stackoverflow.com/a/19593849/6830396

**Comment by kyleli666** [comment]
@GnaneshKunal Thank you.
I'm not using nginx, but AWS ALB formed by Ingress in k8s.
I don't have issue on version 1.11.1 with my proxy setup, is there something new need attetion from 1.12?
Here is my ingress declaration,
```
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: redisinsight-ingress
  namespace: manage-tool
  annotations:
    alb.ingress.kubernetes.io/load-balancer-name: qa-manage-tool
    alb.ingress.kubernetes.io/group.name: manage-tool
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 83}]'
    alb.ingress.kubernetes.io/backend-protocol: HTTP
  labels:
    app: redisinsight
spec:
  ingressClassName: alb
  rules:
    - host: "*.mydomain.com"
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: redisinsight-service
                port:
                  number: 80
```

**Comment by GnaneshKunal** [comment]
@kyleli666 

Thank you very much for the manifest file. Does the setup work in `1.12.0` ? The current latest one is `1.12.1`.

**Comment by kyleli666** [comment]
@GnaneshKunal from my experience,  the setup also does not work in 1.12.0.

**Comment by GnaneshKunal** [comment]
@kyleli666 

Okay. The setup worked till 1.11.1.


Actually we did upgrade core django libraries in 1.12.1 but the libraries used were same in both 1.11.1 and 1.12.0. So I need to figure out the changes. Checking out the issue.

**Comment by GnaneshKunal** [comment]
@kyleli666 

Thank you very much for verifying. Would require your help for testing some builds if needed.

**Comment by GnaneshKunal** [comment]
Hey @kyleli666,

We have a test build. Would be happy if it works for you.

Here's the test image: `redislabs/redisinsight:ri-trusted-origins`


Do pass a new environment variable `RITRUSTEDORIGINS` and set the variable to the domain name/IP of where the service is hosted. Suppose there's a reverse proxy at `localhost:9000`, the `RITRUSTEDORIGINS` value would be `http://localhost:9000`. Notice that a scheme `http` or `https` should also be passed. If you want to pass more than one trusted origin, you can pass them comma separated.

Please let us know if this works for you.

**Comment by ironsalsa** [comment]
I experienced this issue with 1.12.1 as well in GKE Kubernetes behind the NGINX ingress proxy with traffic routed publicly over Cloudflare Argo Tunnels. Using redislabs/redisinsight:ri-trusted-origins and adding the RITRUSTEDORIGINS env added to the pod resolved the issue when using a public https URL.

I'll watch this and see when that is officially added!

**Comment by kyleli666** [comment]
Hi @GnaneshKunal It works for me, too~

**Comment by nifr** [comment]
I just ran into the same issue with `v1.12.1`.

```
HTTP 403
{"detail":"CSRF Failed: Origin checking failed - https://redis.example.tld does not match any trusted origins."}
```

A temporary switch to docker image `docker.io/redislabs/redisinsight:ri-trusted-origins` and setting `RITRUSTEDORIGINS=https://redis.example.tld` resolved the CORS issue.

It would be really awesome to get this fix merged before the next release :)

Thanks for your quick response on this issue @GnaneshKunal

**Comment by GnaneshKunal** [comment]
Hey,


`RITRUSTEDORIGINS` has been released in `1.13.0`. Please check - https://docs.redis.com/latest/ri/using-redisinsight/proxy/#trusted-origins.


Closing this issue.
